homeassistant/components/motionblinds_ble/config_flow.py

```diff
@@ -0,0 +1,222 @@
+                    if type(e) in EXCEPTION_MAP
+                    else str(type(e))
+                }
+                return self.async_show_form(
+                    step_id="user", data_schema=CONFIG_SCHEMA, errors=errors
+                )
+            return await self.async_step_confirm()
+
+        return self.async_show_form(
+            step_id="user", data_schema=CONFIG_SCHEMA, errors=errors
+        )
+
+    async def async_step_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm a single device."""
+        if user_input is not None:
+            self._blind_type = user_input[CONF_BLIND_TYPE]
+
+            assert self._discovery_info is not None
+
+            return self.async_create_entry(
+                title=str(self._display_name),
+                data={
+                    CONF_ADDRESS: self._discovery_info.address,
+                    CONF_LOCAL_NAME: self._discovery_info.name,
+                    CONF_MAC_CODE: self._mac_code,
+                    CONF_BLIND_TYPE: self._blind_type,
+                },
+            )
+
+        return self.async_show_form(
+            step_id="confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_BLIND_TYPE): SelectSelector(
+                        SelectSelectorConfig(
+                            options=[
+                                blind_type.value for blind_type in MotionBlindType
+                            ],
+                            translation_key=CONF_BLIND_TYPE,
+                            mode=SelectSelectorMode.DROPDOWN,
+                        )
+                    )
+                }
+            ),
+            description_placeholders={"display_name": self._display_name},
+        )
+
+    async def async_discover_motionblind(self, mac_code: str) -> None:
+        """Discover MotionBlinds initialized by the user."""
+        if not is_valid_mac(mac_code):
+            _LOGGER.error("Invalid MAC code: %s", mac_code.upper())
+            raise InvalidMACCode()
+
+        count = bluetooth.async_scanner_count(self.hass, connectable=True)
+        if count == 0:
+            self.hass.async_create_task(
+                self.hass.config_entries.flow.async_configure(flow_id=self.flow_id)
+            )
+            _LOGGER.error("No bluetooth adapter found")
+            raise NoBluetoothAdapter()
+
+        bleak_scanner = bluetooth.async_get_scanner(self.hass)
+        devices = await bleak_scanner.discover()
+
+        if len(devices) == 0:
+            _LOGGER.error("Could not find any bluetooth devices")
+            raise NoDevicesFound()
+
+        motion_device: BLEDevice | None = next(
+            (
+                device
+                for device in devices
+                if device
+                and device.name
+                and f"MOTION_{mac_code.upper()}" in device.name
+            ),
+            None,
+        )
```

Comment:
I think you can simplify this by using `bluetooth.async_ble_device_from_address`.

Reply:
Disregard, I didn't notice you where selecting the device by name and not by mac address.